core/camel-support/src/main/java/org/apache/camel/support/processor/UnmarshalProcessor.java

```diff
@@ -55,13 +55,16 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
         InputStream stream = null;
         Object result = null;
         try {
-            stream = exchange.getIn().getMandatoryBody(InputStream.class);
+            final Message in = exchange.getIn();
+            final Message out = exchange.getOut();
```

Comment:
Do not call getOut() to early as it will create an empty message, and cause the exchange to have this empty message as the response, even if the message body is null and allow null enabled. So move this out to later where its first in use